src/xg.hpp

```diff
@@ -245,6 +255,9 @@ class XG {
     /// Extract a particular thread by name. Name may not be empty.
     /// TODO: Actually implement name storage for threads, so we can easily find a thread in the graph by name.
     thread_t extract_thread(const string& name) const;
+
+    thread_t extract_thread(xg::XG::ThreadMapping node, int64_t offset, int64_t max_length);
```

Comment:
This method doesn't extract a thread by name; can you add a new description for it?